src/binary-writer.cc

```diff
@@ -945,16 +939,13 @@ static Result write_module(Context* ctx, const Module* module) {
     }
   }
 
-  return ctx->stream.result;
+  return ctx->stream.result();
 }
 
 Result write_binary_module(Writer* writer,
                            const Module* module,
                            const WriteBinaryOptions* options) {
-  Context ctx;
-  ctx.options = options;
-  ctx.log_stream = options->log_stream;
-  init_stream(&ctx.stream, writer, ctx.log_stream);
+  Context ctx(writer, options);
   return write_module(&ctx, module);
```

Comment:
Yeah, I had a refactor in a earlier commit (making this whole thing into a class), but this makes for a simpler change, I think.